src/coreclr/jit/lsra.cpp

```diff
@@ -6353,7 +6354,7 @@ void LinearScan::insertUpperVectorSave(GenTree*     tree,
     bool spillToMem = refPosition->spillAfter;
     assert(spillReg != REG_NA);
 #else
-    bool spillToMem = (spillReg == REG_NA);
+    bool spillToMem = (spillReg == REG_NA) || (genActualType(varDsc->TypeGet()) == TYP_SIMD64);
```

Comment:
This should not have the check for `TYP_SIMD64`. I am assuming this was added just as a workaround until you find the right solution. Basically, in below portion of the code, you need to set `allocate = false` always for `TYP_SIMD64` because we decided to spill it on stack. Also, the notion of `isPartiallySpilled` is not applicable here, because it is "completely spilled". Until now, we used `isPartiallySpilled`, mainly to know if we have already partially spilled (created RefPositions for save/restore). You might have to use something similar except that it won't be called `isPartiallySpilled` but instead would be `isSpilled` or a better name variable. I just don't want to reuse `isPartiallySpilled` for `TYP_SIMD64` because it is not what we are doing (not spilling it partially).

https://github.com/dotnet/runtime/blob/53bb518ea482a5960577b151c11cc1297e5dea6d/src/coreclr/jit/lsra.cpp#L5067-L5096